openfl/callbacks/callback_list.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from typing import List
+
+from openfl.callbacks.callback import Callback
+from openfl.callbacks.memory_profiler import MemoryProfiler
+
+
+class CallbackList(Callback):
+    """An ensemble of callbacks.
+
+    This class allows multiple callbacks to be used together, by sequentially
+    calling each callback's respective methods.
+
+    Attributes:
+        callbacks: A list of `openfl.callbacks.Callback` instances.
+        add_memory_profiler: If True, adds a `MemoryProfiler` callback to the list.
+        tensor_db: Optional `TensorDB` instance of the respective participant.
+            If provided, callbacks can access TensorDB for various actions.
+        params: Additional parameters saved for use within the callbacks.
+    """
+
+    def __init__(
+        self,
+        callbacks: List[Callback],
```

Comment:
`typing.List` has been deprecated as of python 3.9. [Ref.](https://docs.python.org/3/library/typing.html#typing.List) 
`list` can be used directly.

Reply:
Yay, 3.9 is minimum now